sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -258,6 +287,21 @@ Scheduler::Scheduler() {
                      /*PropList=*/{}));
 }
 
+Scheduler::~Scheduler() {
+  // By specification there are several possible sync points: buffer
+  // destruction, wait() method of a queue or event. Stream doesn't introduce
+  // any synchronization point. It is guaranteed that stream is flushed and
+  // resources are released only if one of the listed sync points was used for
+  // the kernel. Otherwise resources for stream will not be released, issue a
+  // warning in this case.
+  std::lock_guard<std::mutex> lock(StreamBuffersPoolMutex);
+  if (!StreamBuffersPool.empty())
+    std::cerr
```

Comment:
Please do not print to standard output streams as it can be unexpected by customers. I suggest replacing it with an assert or printing only if pi trace is enabled. Also it would be more safe to print using printf to avoid problems with ordering of global objects destruction(Scheduler and IIRC cerr are global objects).

Reply:
Assertion won't available in release build. I believe the user won't ever see it. Thus, it should be either an output with PI tracing or an exception.